### PR TITLE
example-posix brings up an HTTP POST /add handler

### PR DESCRIPTION
**Pull request by mhutchinson**
This makes this example consistent in API with the other examples, and allows the implementation to be load tested via the hammer

### CODE REVIEW

File: cmd/example-posix/README.md [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+export LOG_PRIVATE_KEY="PRIVATE+KEY+example.com/log/testdata+33d7b496+AeymY/SZAX0jZcJ8enZ5FY1Dz+wTML2yWSkK+9DSF3eg"
+export LOG_PUBLIC_KEY="example.com/log/testdata+33d7b496+AeHTu4Q3hEIMHNqc6fASMsq3rKNx280NI+oO5xCFkkSx"
+
+# Initialize a new log
+export LOG_DIR=/tmp/mylog2
+go run ./cmd/example-posix --storage_dir=${LOG_DIR} --initialise --v=2
+```
+
+Then run the hammer against the log.
+In this example, we're running 32 writers agains the log to add 128 new leaves within 1 minute.
```
Comment (AlCutter) [inline cmd/example-posix/README.md:19]:
```suggestion
In this example, we're running 32 writers against the log to add 128 new leaves within 1 minute.
```

File: cmd/example-posix/README.md [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+## Using in the hammer
+
+First bring up a new log:
+```shell
+export LOG_PRIVATE_KEY="PRIVATE+KEY+example.com/log/testdata+33d7b496+AeymY/SZAX0jZcJ8enZ5FY1Dz+wTML2yWSkK+9DSF3eg"
+export LOG_PUBLIC_KEY="example.com/log/testdata+33d7b496+AeHTu4Q3hEIMHNqc6fASMsq3rKNx280NI+oO5xCFkkSx"
+
+# Initialize a new log
+export LOG_DIR=/tmp/mylog2
+go run ./cmd/example-posix --storage_dir=${LOG_DIR} --initialise --v=2
```
Comment (AlCutter) [inline cmd/example-posix/README.md:15]: Worth explicitly setting `--listen`?

File: cmd/example-posix/main.go [hunk 3]
```diff
@@ -12,17 +12,17 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
-// example-posix is a command line tool for adding entries to a local
+// example-posix runs a web server that allows new entries to be POSTed to
```
Comment (AlCutter) [inline cmd/example-posix/main.go:15]: comment missing an "a" at the end?

File: cmd/example-posix/main.go [hunk 4]
```diff
@@ -100,27 +100,16 @@ func main() {
 		return posix.WriteCheckpoint(*storageDir, n)
 	}
 	if *initialise {
+		if err := os.RemoveAll(*storageDir); err != nil {
```
Comment (AlCutter) [inline cmd/example-posix/main.go:103]: This seems bold :)
I guess it's just an example, but probably don't want people to copy this behaviour!

**Review by AlCutter — APPROVED** [review]

File: cmd/example-posix/README.md (hunk 2 above)
Comment (mhutchinson) [reply cmd/example-posix/README.md]: Sure.

File: cmd/example-posix/README.md (hunk 1 above)
Comment (mhutchinson) [reply cmd/example-posix/README.md]: Done.

File: cmd/example-posix/main.go (hunk 3 above)
Comment (mhutchinson) [reply cmd/example-posix/main.go:15]: Done.

File: cmd/example-posix/main.go (hunk 4 above)
Comment (mhutchinson) [reply cmd/example-posix/main.go:103]: Yeah I've got mixed feelings about it. I can make it fail instead if there are contents in the directory?

Comment (AlCutter) [reply cmd/example-posix/main.go:103]: Yeah, seems safer

**Review by AlCutter — APPROVED** [review]

**Pull request merged**